binance/threaded_stream.py

```diff
@@ -73,6 +76,15 @@ def stop(self):
         if not self._running:
             return
         self._running = False
-        self._loop.call_soon(asyncio.create_task, self.stop_client())
+        if self._client and self._loop and not self._loop.is_closed():
+            try:
+                future = asyncio.run_coroutine_threadsafe(
+                    self.stop_client(), 
+                    self._loop
+                )
+                future.result(timeout=5)  # Add timeout to prevent hanging
```

Comment:
@pcriadoperez did you face issues with hangs here?

Reply:
Not in the current tests. But I'll add a test for start, stop start again just in case.